### DISCUSSION TITLE
Multiple SubjectConfirmation Bug

### DISCUSSION THREAD

**Discussion by Bluebite9** [bergie/passport-saml#693] [Q&A]
Hello,

I am getting "Unable to process multiple SubjectConfirmations in SAML assertion" in the callback function after login.

`this.router.post('/login/callback',
            passport.authenticate('saml', config.saml.options),
            (req: express.Request, res: express.Response, next: NextFunction) => {
                return res.redirect(config.redirectAfterSuccess);
            });`

This is the callback function. I noticed an opened issue #339 and also that in the node-saml library it was fixed. Is it fixed in the passport-saml as well?

`<saml2:SubjectConfirmation Method="urn:oasis:names:tc:SAML:2.0:cm:bearer">
                <saml2:SubjectConfirmationData InResponseTo="_3efc23bad482ec2863a3" NotOnOrAfter="2022-06-02T08:08:51.937Z" Recipient="http://localhost:8081/sso/login/callback"/>
            </saml2:SubjectConfirmation>
            <saml2:SubjectConfirmation Method="urn:oasis:names:tc:SAML:2.0:cm:bearer">
                <saml2:SubjectConfirmationData InResponseTo="_3efc23bad482ec2863a3" NotOnOrAfter="2022-06-02T08:08:51.937Z" Recipient="https://localhost/oauth2/token"/>
            </saml2:SubjectConfirmation>`

Those are the subject confirmations. I am testing in localhost.

Best regards and thanks for your help!

**Comment by srd90** [comment]
> Is it fixed in the passport-saml as well?

When node-saml/passport-saml 4.0 is released it shall include node-saml/node-saml 4.x. I.e. starting from passport-saml 4.0 core saml functionality for passport-saml 4.x shall be provided by node-saml >= 4.x. node-saml 4.x shall be listed as dependency in passport-saml 4.0's package.json

passport-saml versions < 4.0 do not contain that support at the moment (and node-saml functionality is ”inlined” at passport-saml < 4.0 versions instead of using it from separated package).

**Reply by cjbarth** [reply]
passport-saml@4 has been released @Bluebite9
